server/jobs/job_url_setter.go

```diff
@@ -0,0 +1,47 @@
+package jobs
+
+import "github.com/runatlantis/atlantis/server/events/models"
+
+//go:generate pegomock generate -m --use-experimental-model-gen --package mocks -o mocks/mock_project_job_url_generator.go ProjectJobURLGenerator
+
+// ProjectJobURLGenerator generates urls to view project's progress.
+type ProjectJobURLGenerator interface {
+	GenerateProjectJobURL(p models.ProjectCommandContext) (string, error)
+}
+
+//go:generate pegomock generate -m --use-experimental-model-gen --package mocks -o mocks/mock_project_status_updater.go ProjectStatusUpdater
+
+type ProjectStatusUpdater interface {
```

Comment:
I want us to get away from this practice of exporting interfaces because it creates an unnecessary dependency between packages.  Also it's more semantic in go to always return a concrete type and use an interface on the reader side.

Here's an example of this: https://stackoverflow.com/questions/51031116/export-interfaces-instead-of-structs-from-golang-packages

Reply:
Yeah I like this approach too! I've moved the `JobURLSetter` to `project_command_runner` and left the unexported  interfaces `ProjectJobURLGenerator` and `ProjectStatusUpdater` as it is.